modules/contacts/site.js

```diff
@@ -20,12 +20,57 @@ var delete_contact = function(id, source, type) {
 var add_contact_from_message_view = function() {
     var contact = $('#add_contact').val();
     var source = $('#contact_source').val();
+  
+    if (contact) {
+      Hm_Ajax.request(
+        [
+          { name: 'hm_ajax_hook', value: 'ajax_add_contact' },
+          { name: 'contact_value', value: contact },
+          { name: 'contact_source', value: source },
+        ],
+        function (res) {
+          $('.add_contact_controls').toggle();
+          window.location.reload();
```

Comment:
If you reload the whole page, the other JS operations you do seems unneeded. Why do we need to reload the whole page? Can't we do the updates here with JS?

Reply:
I reload to refresh the session storage, I follow the way they implemented `add_contact_from_message_view`. is there a way to refresh the cache page from session storage without reloading the page?

Reply:
Session storage is available at all times - you don't need to reload the page to get the latest data in session storage. If you need to refresh specific sections on the page, you can use JS functions for this or ajax requests.